packages/domains-table/src/domains-table/index.tsx

```diff
@@ -10,12 +10,15 @@ import { useState, useCallback, useLayoutEffect, useMemo } from 'react';
 import { DomainsTableColumn, DomainsTableHeader } from '../domains-table-header';
 import { domainsTableColumns } from '../domains-table-header/columns';
 import { getDomainId } from '../get-domain-id';
+import { DomainStatusPurchaseActions } from '../utils/resolve-domain-status';
 import { DomainsTableRow } from './domains-table-row';
 import './style.scss';
 
 interface DomainsTableProps {
 	domains: PartialDomainData[] | undefined;
 	isAllSitesView: boolean;
+	dispatch: any;
+	domainStatusPurchaseActions: DomainStatusPurchaseActions< any >;
```

Comment:
I'm not certain how we're going to resolve these types since they reside client side, any input is appreciated.

I've created this type for purchases actions needed but allow them to be passed as props since purchases does not directly relates to domain and share different libraries

Reply:
We can use a generic, React's TSX infers generics too.